src/stactools/sentinel2/product_metadata.py

```diff
@@ -16,6 +16,22 @@ class ProductMetadataError(Exception):
     pass
 
 
+def fix_z_values(coord_values: List[str]) -> List[float]:
+    """Some geometries have a '0' value in the z position
+    of the coordinates. This method detects and removes z
+    postion coordinates. This assumes that in cases where
+    the z value is included, it is included for all coordinates.
+    """
+    if len(coord_values) % 3 == 0:
+        # Check if all of the 3rd position values are '0'
+        if not any(
+            [x
+             for i, x in enumerate(coord_values) if i % 3 == 2 and x != '0']):
```

Comment:
This is hard parse.

```suggestion
        if all(x != '0' for i, x in enumerate(coord_values) if i % 3 == 2):
```

Reply:
I don't think these are equivalent. 

Mine is, if there aren't any 3rd position values that don't equal zero.
Yours is, if all 3rd position values don't equal zero.

I'll rewrite for clarity